source/common/http/header_list_view.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <vector>
+
+#include "envoy/http/header_map.h"
+
+namespace Envoy {
+namespace Http {
+
+class HeaderListView {
+public:
+  using HeaderKeyValuePairWrapper =
+      std::vector<std::pair<std::reference_wrapper<const HeaderString>,
+                            std::reference_wrapper<const HeaderString>>>;
+  HeaderListView(const HeaderMap& header_map);
+  std::vector<std::reference_wrapper<const HeaderString>> keys() const;
+  std::vector<std::reference_wrapper<const HeaderString>> values() const;
```

Comment:
This is pretty inefficient to copy once into this view, and then copy again for keys and values. What is the purpose of this class exactly?

/wait-any

Reply:
There is no purpose for this. Fix it.